Server.Tests/Git/ToolsCommands/GitCloneShould.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Moq;
+using PrincipleStudios.ScaledGitApp.ShellUtilities;
+
+namespace PrincipleStudios.ScaledGitApp.Git.ToolsCommands;
+
+public class GitCloneShould : IClassFixture<GitToolsPowerShellFixture>
+{
+	private readonly GitToolsPowerShellFixture fixture;
+
+	public GitCloneShould(GitToolsPowerShellFixture fixture)
+	{
+		this.fixture = fixture;
+	}
+
+	[Fact]
+	public async Task Issue_a_clone()
+	{
+		var expectedRepository = "https://example.com/.git";
+		var mockFinal = new Mock<IPowerShell>();
+		fixture.MockPowerShellFactory.Setup(ps => ps.Create(null)).Returns(mockFinal.Object);
+
+		var verifyGitClone = SetupGitClone(mockFinal, expectedRepository);
+
+		// By mocking the factory directly, we test the typical DI constructor with working directory setup
+		using var target = fixture.CreateTarget();
+
+		await target.GitClone(expectedRepository);
+
+		verifyGitClone.Verify(Times.Once);
+	}
+
+	internal static VerifiableMock<IPowerShell, Task<PowerShellInvocationResult>> SetupGitClone(Mock<IPowerShell> target, string expectedRepository)
+	{
+		// This is very permissive right now; clone should be bare, and then we don't need the other setups
+		target.Setup(ps => ps.InvokeCliAsync("git", It.IsAny<string[]>())).ReturnsAsync(PowerShellInvocationResultStubs.Empty);
+		target.Setup(ps => ps.InvokeCliAsync("git", "rev-parse", "--abbrev-ref", "HEAD")).ReturnsAsync(PowerShellInvocationResultStubs.WithResults("main"));
+		return target.Verifiable(
+			ps => ps.InvokeCliAsync("git", It.Is<string[]>(args => VerifyCliArgs(args, expectedRepository))),
+			s => s.ReturnsAsync(PowerShellInvocationResultStubs.Empty)
+		);
+	}
+
+	static bool VerifyCliArgs(string[] args, string expectedRepository)
+	{
+		var nonSwitchArgs = args.Where(arg => !arg.StartsWith('-')).ToArray();
+		return nonSwitchArgs.Length == 3 && nonSwitchArgs[0] == "clone" && nonSwitchArgs[1] == expectedRepository && nonSwitchArgs[2] == ".";
```

Comment:
Just double checking, is it always safe to assume there will only be 3 non switch args that will be passed for verification to pass?

Reply:
At this time, yeah; it's pretty hard-coded. That said, I just refactored this in a different branch to use the command pattern, let me double-check this after I back-merge that to here.

Reply:
Added comments to the method header to describe what it's for, but since you can't see that here (and it doesn't show outdated)...
The git command being run is `git clone https://example.com/.git .` with some additional switches like `--no-checkout` and `--quiet`. They need to be in order. There could be other switch args that have parameters that name the remote, but in our case, we don't have that.